pkg/reconciler/revision/background.go

```diff
@@ -200,10 +200,16 @@ func (r *backgroundResolver) processWorkItem(item *workItem) {
 		r.queue.Done(item)
 	}()
 
+	// We need to acquire the result under lock since it's theoretically possible
+	// for a Clear to race with this and try to delete the result from the map.
+	r.mu.Lock()
```

Comment:
can be readlock?

Reply:
done, also this reminded me to add a nil check here for the exact race the comment is worried about 😊